textworld/generator/inform7/world2inform7.py

```diff
@@ -393,7 +394,8 @@ def generate_inform7_source(game, seed=1234, use_i7_description=False):
         remove the list of containers from L;
         remove the list of supporters from L;
         remove the list of doors from L;
-        say "There is [L with indefinite articles] on the floor.";
+        if the number of entries in L is greater than 0:
+            say "There is [L with indefinite articles] on the floor.";
```

Comment:
While at it, I've fixed this bug. Right now, when there is no object on the ground it displays: `"There is  on the floor."`